### ISSUE TITLE
Full or partial of detected object 

### ISSUE THREAD

**Issue by Auth0rM0rgan** [ultralytics/yolov5#2011]
## ❔Question
Hey @glenn-jocher, 

I'm looking for a way to understand that if the detected object from yolov5 is the full object or is the part of it. For example, I'm interested in detecting cars and I want to know if the detected car has appeared fully or partially. I have tried to find the keypoints and descriptors with SIFT and did the image matching with my reference image but the result wasn't reliable! is there any way to do that? or is there any way to show only the bounding box when the car detected completely, not part of the car?

Thanks in advance!

**Comment by glenn-jocher** [comment]
@Auth0rM0rgan the concept is very simple. The model learns what it's trained on, end of story. If you want it to ignore partial objects, your dataset must be free of labelled partial objects.

**Comment by Ownmarc** [comment]
If you expect the "full" car to almost always produce a certain shape of bounding box, lets say a square shape (width == height), you could check for that with a tolerance and if its not the shape of a full car, then you know its a part of a car.
If your "half car" is always on the edge of your image, you could also check for the location of the box. If its really near the edge and has a shape that doesn't look like the "car" shape, you could also implement that.

The best approach, I think, would be to create a new dataset with what ever the model should output. So if the car is in fact "half a car", then its the label it should get and you could give it its own class. Make sure your annotations are coherent with each other and the model should give you the result you want

**Comment by Auth0rM0rgan** [comment]
@glenn-jocher, @Ownmarc Thanks for the suggestions! maybe I wasn't clear enough! I meant I want to detect all the possible cars are appearing in the video with any shapes and draw only bounding boxes for the cars which appeared with the full shape. So creating a dataset without partial objects is not a solution since I want to have the location of partial objects but only illustrate the full objects! and creating a dataset by assigning a new class to partial objects is not a solution, because the dataset is huge! Is there any other solution?

Thanks!

**Comment by glenn-jocher** [comment]
@Auth0rM0rgan there's always a solution. In your case though the solution would require some customization. You require a model update for one additional output dimension, i.e. 'area', spanning from 0-1 (fully contained vehicles are 1). You can do this two ways:

- Create a dataset of only fully contained vehicles (no partial vehicles at all). Then after augmentation in the dataloader compute the remaining area of the vehicle in the frame and add that information to your labels (and update loss function).
- Create a dataset of fully and partially contained vehicles, and manually assign the area value for each label (and update loss function accordingly).

**Comment by Auth0rM0rgan** [comment]
Hey @glenn-jocher, 
Thanks for the suggestion! I tried it but I couldn't succeed yet! Would you please elaborate more on how to compute the remaining area and update the loss function? 

Thanks!

**Comment by glenn-jocher** [comment]
@Auth0rM0rgan sorry, no, no time to additionally support this beyond my previous comments.

**Comment by nsenel** [comment]
Hey @Auth0rM0rgan,

Have you find a solution?

**Comment by leyangjin** [comment]
Hey @Auth0rM0rgan 

Have you find a solution?
